test/net/sourceforge/kolmafia/persistence/ItemFinderTest.java

```diff
@@ -0,0 +1,833 @@
+package net.sourceforge.kolmafia.persistence;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import net.sourceforge.kolmafia.AdventureResult;
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.KoLConstants;
+import net.sourceforge.kolmafia.KoLConstants.MafiaState;
+import net.sourceforge.kolmafia.KoLmafia;
+import net.sourceforge.kolmafia.StaticEntity;
+import net.sourceforge.kolmafia.objectpool.ItemPool;
+import net.sourceforge.kolmafia.persistence.ItemFinder.Match;
+import net.sourceforge.kolmafia.request.CharPaneRequest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ItemFinderTest {
+
+  @BeforeEach
+  private void itemFinderSetup() {
+    // Simulate logging out and back in again.
+    KoLCharacter.reset("");
+    KoLCharacter.reset("item finder user");
+    // Reset preferences to defaults.
+    KoLCharacter.reset(true);
+
+    // Not in error state
+    StaticEntity.setContinuationState(MafiaState.CONTINUE);
+
+    // Default is not in Ronin or Hardcore
+    KoLCharacter.setRonin(false);
+    KoLCharacter.setHardcore(false);
+    CharPaneRequest.setCanInteract(true);
+
+    KoLConstants.inventory.clear();
```

Comment:
The log out / log in above should have already cleared the inventory (and everything else). There shouldn't be any need to do it twice.

Some work was done (mostly in MaximizerTests) in an attempt to avoid using `reset` between tests, and do the smallest amount of setup / teardown possible. This might be worth looking into, but I don't think it's required at the moment.

Reply:
I want to clear everything before the whole suite runs, so the login/etc. could go in a @BeforeAll.
I want to be free to set up various item lists in each (group of) tests, so reseting items lists (and hardcore/ronin/interaction) could go in @BeforeEach or @AfterEach.
Ditto for continuation state; don't want to stay in ERROR, for example.

I'll try splitting into @BeforeAll and @BeforeEach, at least, to minimize the mega-reset

Reply:
Made @BeforeAll and @BeforeEach, as discussed.
Reset exactly the two Preferences that ItemFinder uses in the @BeforeEach.